mygit.rb

```diff
@@ -0,0 +1,365 @@
+         Dir.chdir "#{repo}/.#{repo}"
+         if File.file?('gitconfig.txt')
+                $LOG.debug("<MyGit> : <ConfigExist?> : The config file exists")
+                Dir.chdir('../..')
+                if ARGV.include?('-f')
+                        $LOG.debug("<MyGit> : <ConfigExist?> : Forcefully creating a new config")
+                        ConfigGit()
+                        return                        
+                end
+                puts "<MyGit> : Config Exists already \n use -f tag to force a new config"
+                $LOG.debug("<MyGit> : <ConfigExist?> : Config Already Exists!")
+         else
+                $LOG.debug("<MyGit> : <ConfigExist?> : Calling the ConfigGit function to configure the user's git")
+                Dir.chdir('../..')
+                ConfigGit()
+        
+         end
+
+         
+end
+
+
+def ReturnGitCredentials()
+        $LOG.debug("<MyGit> <ReturnGitCredentials> : Checking if credentials are stored")
+        repo=getreponame()
+        Dir.chdir(repo)
+        gitconfig = readhidden(repo,"gitconfig.txt")
+        gitconfig = gitconfig.split("\n")
+
+        return gitconfig
+end
+
+
+def ConfigGit()
+        
+        $LOG.debug("<MyGit> <ConfigGit> : Asking for the user`s git credentials through console")
+        print "Enter your GitHub username: "
+        username = STDIN.gets.chomp
+        print "Enter your GitHub Passowrd: "
+        password = STDIN.noecho(&:gets).chomp
+        puts "\n"
+        gitconf = username+"\n"+password
+        repo = getreponame()
+        Dir.chdir(repo)
+        $LOG.debug("<MyGit> <ConfigGit> : Writing to gitconfig file")
+        writetofile("gitconfig.txt",gitconf,repo)
+        $LOG.debug("<MyGit> <ConfigGit> : Git credentials have been configured")
+        puts "<MyGit> : Git Credentials have been successfully configured"
+end
+
+
+def octoKitInit()
+        $LOG.debug("<MyGit> <octoKitInit>  : Initializing Octokit API for git")
+        gitconfig = ReturnGitCredentials()
+        login = gitconfig[0]
+        password = gitconfig[1]
+        client = Octokit::Client.new(:login => login , :password => password)
+        $LOG.debug("<MyGit> <octoKitInit> : Successfully logged into Octokit")
+        return client
+end
+
+def GitExist?
+        $LOG.debug("<MyGit> : <GitExist?> : Checking if the user has git installed")
+        git_path = `which git`
+        if git_path.length > 0
+                $LOG.debug("<MyGit> : <GitExist?> : Git exists")
+                return true
+
+        else
+                $LOG.debug("<MyGit> : <GitExist?> : Git isnt installed, requesting the user to install git")
+                puts "<MyGit> : <Error> : git isn`t installed on your system, kindly install and run the app again!"
+                return false
+        end
+        
+end     
+
+
+def getreponame
+        $LOG.debug("<MyGit> : <getreponame> : Asking for reponame")
+        repo = File.open('repo.txt', 'r').read
```

Comment:
File is not closed , also reusing file variable to assign string is bad practice.

For line by line reading of file 
```ruby
File.open("path","r").each do |line|
end
```

each_with_index also exists if you need only the second line .

Reply:
> For line by line reading of file

would this be necessary even if my file contains only a single line? as is the case with my file, which contains only the repo directory.